lib/resty/acme/client.lua

```diff
@@ -84,16 +80,30 @@ function _M.new(conf)
     self.challenge_handlers[c] = handler.new(self.storage)
   end
 
-  local account_thumbprint, err = util.thumbprint(self.account_pkey)
-  if err then
-    return nil, "failed to calculate thumbprint: " .. err
+  if conf.account_key then
+    local account_pkey = openssl.pkey.new(conf.account_key)
+    self.account_pkey = account_pkey
+    local account_thumbprint, err = util.thumbprint(account_pkey)
+    if err then
+      return nil, "failed to calculate thumbprint: " .. err
+    end
+    self.account_thumbprint = account_thumbprint
   end
 
-  self.account_thumbprint = account_thumbprint
-
   return self
 end
 
+function _M:load_account_key(account_key)
```

Comment:
let's make this function a local and set in _M, so we can reuse in new(), also maybe call it set_account_key?

```lua
local function set_account_key(self, account_key)
....
end

function _M.new(self)
...
if conf.account_key then
    local ok, err = set_account_key(self, conf.account_key)
    if err then
         return nil, errr
    end
...
end


_M.set_account_key = set_account_key
```